components/tikv_kv/src/lib.rs

```diff
@@ -456,10 +458,24 @@ pub unsafe fn destroy_tls_engine<E: Engine>() {
 pub fn snapshot<E: Engine>(
     engine: &E,
     ctx: SnapContext<'_>,
+    quota_limiter: Option<Arc<QuotaLimiter>>,
 ) -> impl std::future::Future<Output = Result<E::Snap>> {
+    let start_time = if let Some(quota_limiter) = &quota_limiter {
+        quota_limiter.get_now_time()
+    } else {
+        Box::new(Instant::now())
+    };
+
     let (callback, future) =
         tikv_util::future::paired_must_called_future_callback(drop_snapshot_callback::<E>);
     let val = engine.async_snapshot(ctx, callback);
+    if let Some(quota_limiter) = quota_limiter {
```

Comment:
two issues here:
1) async_snapshot is not actually finished at L472, the duration collected is wrong. 
2) cannot use std::thread::sleep as it will make the whole thread sleep.